packages/eas-cli/src/update/configure.ts

```diff
@@ -0,0 +1,291 @@
+  return workflow === Workflow.GENERIC
+    ? DEFAULT_BARE_RUNTIME_VERSION
+    : DEFAULT_MANAGED_RUNTIME_VERSION;
+}
+
+function isRuntimeEqual(
+  runtimeVersionA: NonNullable<ExpoConfig['runtimeVersion']>,
+  runtimeVersionB: NonNullable<ExpoConfig['runtimeVersion']>
+): boolean {
+  if (typeof runtimeVersionA === 'string' && typeof runtimeVersionB === 'string') {
+    return runtimeVersionA === runtimeVersionB;
+  } else if (typeof runtimeVersionA === 'object' && typeof runtimeVersionB === 'object') {
+    return runtimeVersionA.policy === runtimeVersionB.policy;
+  } else {
+    return false;
+  }
+}
+
+/**
+ * Make sure the `app.json` is configured to use EAS Updates.
+ * This does a couple of things:
+ *   - Ensure update URL is set to the project EAS endpoint
+ *   - Ensure runtimeVersion is defined for both or individual platforms
+ *   - Output the changes made, or the changes required to make manually
+ */
+export async function ensureEASUpdatesIsConfiguredInExpoConfigAsync({
+  exp,
+  projectId,
+  projectDir,
+  platform,
+  workflows,
+}: {
+  exp: ExpoConfig;
+  projectId: string;
+  projectDir: string;
+  platform: RequestedPlatform;
+  workflows: Record<Platform, Workflow>;
+}): Promise<{ projectChanged: boolean; exp: ExpoConfig }> {
+  const modifyConfig: Partial<ExpoConfig> = {};
+
+  if (exp.updates?.url !== getEASUpdateURL(projectId)) {
+    modifyConfig.updates = { url: getEASUpdateURL(projectId) };
+  }
+
+  let androidRuntimeVersion = exp.android?.runtimeVersion ?? exp.runtimeVersion;
+  let iosRuntimeVersion = exp.ios?.runtimeVersion ?? exp.runtimeVersion;
+
+  if (
+    (['all', 'android'].includes(platform) && !androidRuntimeVersion) ||
+    (['all', 'ios'].includes(platform) && !iosRuntimeVersion)
+  ) {
+    androidRuntimeVersion = androidRuntimeVersion ?? getDefaultRuntimeVersion(workflows.android);
+    iosRuntimeVersion = iosRuntimeVersion ?? getDefaultRuntimeVersion(workflows.ios);
+
+    if (platform === 'all' && isRuntimeEqual(androidRuntimeVersion, iosRuntimeVersion)) {
+      modifyConfig.runtimeVersion = androidRuntimeVersion;
+    } else {
+      if (['all', 'android'].includes(platform)) {
+        modifyConfig.runtimeVersion = undefined;
+        modifyConfig.android = { runtimeVersion: androidRuntimeVersion };
+      }
+      if (['all', 'ios'].includes(platform)) {
+        modifyConfig.runtimeVersion = undefined;
+        modifyConfig.ios = { runtimeVersion: iosRuntimeVersion };
+      }
+    }
+  }
+
+  if (Object.keys(modifyConfig).length === 0) {
+    return { exp, projectChanged: false };
+  }
+
+  // NOTE(cedric): might be better with a mergeDeep method, or handle in `modifyConfigAsync`
+  const mergedExp = {
+    runtimeVersion: modifyConfig.runtimeVersion ?? exp.runtimeVersion,
+    updates: { ...exp.updates, ...modifyConfig.updates },
+    android: {
+      ...exp.android,
+      ...modifyConfig.android,
+    },
+    ios: {
+      ...exp.ios,
+      ...modifyConfig.ios,
+    },
+  };
```

Comment:
Would be cool if this was split into a different function that returned `mergedExp` so you could easily test the results and ensure the clobbering always works as expected. I could see a situation where someone changes the `modifyConfig` to define a value like `runtimeVersion: undefined` which then removes the `runtimeVersion` altogether.

Reply:
Setting something to `undefined` does indeed remove that property, at least when using `JSON.parse` (there is no `undefined` type in JSON).

I'll split it regardless for future testing.